docs/en/reference/dql-doctrine-query-language.rst

```diff
@@ -608,6 +608,38 @@ You can also nest several DTO :
 	
 Note that you can only pass scalar expressions or other Data Transfer Objects to the constructor.
 
+If you use your data-transfer objects for multiple query but not ever the same parameters, you can use named arguments or variadic arguments, add ``WithNamedArguments`` to your Dto :
+
+.. code-block:: php
+
+    <?php
+
+    use Doctrine\ORM\WithNamedArguments;
+
+    class CustomerDTO implements WithNamedArguments
+    {
+        public function __construct(public string|null $name = null, public string|null $email = null, public string|null $city = null, public mixed|null $value = null)
+        {
+        }
+    }
+
+And then you can select the fields you want in order you want :
+
+.. code-block:: php
+
+    <?php
+    $query = $em->createQuery('SELECT NEW CustomerDTO(a.city, c.name) FROM Customer c JOIN c.address a');
```

Comment:
Wait so what happens here? Which fields of the DTO are supposed to get hydrated? Hydrating `CustomerDTO::$name` with a city feels wrong :thinking:

Reply:
No, it's exactly the case I want, As the args are named, it's city and name from the Dto witch are filled .
Maybe I have to print the Dto Output for better understand

Reply:
Yes, and also, you could explain what is going to happen: `And then you can select the fields you want in the order you want, and the ORM will try to match argument names with the selected field names`. If you do that, you should probably explain what happens if one of the selected field matches, but not the other. Or what happens if we do `SELECT NEW CustomerDTO(a.city, b.city) FROM …`